exchanges/btcmarkets/btcmarkets_wrapper.go

```diff
@@ -514,124 +535,109 @@ func (b *BTCMarkets) GetFeeByType(feeBuilder *exchange.FeeBuilder) (float64, err
-	order.FilterOrdersBySide(&orders, req.OrderSide)
-	return orders, nil
+	order.FilterOrdersByType(&resp, req.OrderType)
+	order.FilterOrdersByTickRange(&resp, req.StartTicks, req.EndTicks)
+	order.FilterOrdersBySide(&resp, req.OrderSide)
+	return resp, nil
 }
 
 // GetOrderHistory retrieves account order information
 // Can Limit response to specific order status
 func (b *BTCMarkets) GetOrderHistory(req *order.GetOrdersRequest) ([]order.Detail, error) {
+	var resp []order.Detail
+	var tempResp order.Detail
+	var tempArray []string
 	if len(req.Currencies) == 0 {
-		return nil, errors.New("requires at least one currency pair to retrieve history")
-	}
-
-	var respOrders []Order
-	for i := range req.Currencies {
-		resp, err := b.GetOrders(req.Currencies[i].Base.String(),
-			req.Currencies[i].Quote.String(),
-			200,
-			0,
-			true)
+		orders, err := b.GetOrders("", "", "")
 		if err != nil {
-			return nil, err
+			return resp, err
+		}
+		for x := range orders {
+			tempArray = append(tempArray, orders[x].OrderID)
 		}
-		respOrders = append(respOrders, resp...)
 	}
-
-	var orders []order.Detail
-	for i := range respOrders {
-		var side order.Side
-		if strings.EqualFold(respOrders[i].OrderSide, order.Ask.String()) {
-			side = order.Sell
-		} else if strings.EqualFold(respOrders[i].OrderSide, order.Bid.String()) {
-			side = order.Buy
+	for y := range req.Currencies {
+		orders, err := b.GetOrders(b.FormatExchangeCurrency(req.Currencies[y], asset.Spot).String(), "", "")
+		if err != nil {
+			return resp, err
 		}
-		orderDate := time.Unix(int64(respOrders[i].CreationTime), 0)
-		orderType := order.Type(strings.ToUpper(respOrders[i].OrderType))
-
-		openOrder := order.Detail{
-			ID:              strconv.FormatInt(respOrders[i].ID, 10),
-			Amount:          respOrders[i].Volume,
-			Exchange:        b.Name,
-			RemainingAmount: respOrders[i].OpenVolume,
-			OrderDate:       orderDate,
-			OrderSide:       side,
-			OrderType:       orderType,
-			Price:           respOrders[i].Price,
-			Status:          order.Status(respOrders[i].Status),
-			CurrencyPair: currency.NewPairWithDelimiter(respOrders[i].Instrument,
-				respOrders[i].Currency,
-				b.GetPairFormat(asset.Spot, false).Delimiter),
+		for z := range orders {
+			tempArray = append(tempArray, orders[z].OrderID)
 		}
-
-		for j := range respOrders[i].Trades {
-			tradeDate := time.Unix(int64(respOrders[i].Trades[j].CreationTime), 0)
-			openOrder.Trades = append(openOrder.Trades, order.TradeHistory{
-				Amount:      respOrders[i].Trades[j].Volume,
-				Exchange:    b.Name,
-				Price:       respOrders[i].Trades[j].Price,
-				TID:         respOrders[i].Trades[j].ID,
-				Timestamp:   tradeDate,
-				Fee:         respOrders[i].Trades[j].Fee,
-				Description: respOrders[i].Trades[j].Description,
-			})
+	}
+	tempData, err := b.GetBatchTrades(tempArray)
+	if err != nil {
+		return resp, err
+	}
+	for c := range tempData.Orders {
+		if tempData.Orders[c].OpenAmount != 0 {
```

Comment:
I'm not sure that this fixes things properly. We get the order status back, we just want historical orders, so anything that isn't `Accepted`, `Placed` and `Partially Matched` will work
src: https://api.btcmarkets.net/doc/v3#tag/order_status

Orders that are partially/cancelled would still be classified as `Filled` as per the nit below